### PR TITLE
Create "Is this task name unique? Res: True/False" endpoint

### PR DESCRIPTION
**Pull request by davidnmora**
For use to prevent renaming tasks to an existing name.
CORRESPONDS TO THIS SNICKER PR: https://github.com/polymerteam/gelato/pull/124

### CODE REVIEW

File: ics/v10/views.py [hunk 1]
```diff
@@ -230,6 +230,25 @@ class SimpleTaskSearch(generics.ListAPIView):
   def get_queryset(self):
     return simpleTaskSearch(self.request.query_params)
 
+# tasks/name-exists/${name}
+# Receives a task name "name" param, responds with True if it already exists (else False)
+class TaskNameExists(generics.ListAPIView):
+  def get(self, request):
+    def get_task_display(task):
```
Comment (mayanb) [inline ics/v10/views.py]: can we get rid of get_task_display since we no longer need it?

File: ics/v10/views.py [hunk 2]
```diff
@@ -230,6 +230,25 @@ class SimpleTaskSearch(generics.ListAPIView):
   def get_queryset(self):
     return simpleTaskSearch(self.request.query_params)
 
+# tasks/name-exists/${name}
+# Receives a task name "name" param, responds with True if it already exists (else False)
+class TaskNameExists(generics.ListAPIView):
```
Comment (mayanb) [inline ics/v10/views.py]: why is this a list api view if we are just returning 1 object?

**Review by mayanb — CHANGES_REQUESTED** [review]

File: ics/v10/views.py [hunk 3]
```diff
@@ -230,6 +230,25 @@ class SimpleTaskSearch(generics.ListAPIView):
...
+      if task.label_index > 0:
+        return "-".join([task.label, str(task.label_index)])
+      return task.label
+
+    team_id = self.request.query_params.get('team_created_by', None)
+    new_name = self.request.query_params.get('name', None)
+    num_matching_names = Task.objects.filter(is_trashed=False, process_type__team_created_by=team_id) \
+      .annotate(name=Case(When(label_index=0, then=F('label')), default=Concat(F('label'), Value('-'), F('label_index')), output_field=models.CharField())) \
+      .filter(Q(name=new_name) | Q(custom_display=new_name)).count()
+    # queryset = Task.objects.filter(is_trashed=False, process_type__team_created_by=team_id)
```
Comment (mayanb) [inline ics/v10/views.py]: can we delete this commented out line?

**Review by mayanb — APPROVED** [review]

**Comment by ishitap** [comment]
I can't actually get the validation to work. Meaning I can rename the task to something that is already in use...

...actually this seems to be an issue with my db???? hang on

yes it was an issue with gelato

**Pull request merged**